### ISSUE TITLE
Support STAC spec version 1.1.0

### ISSUE THREAD

**Issue by jsignell** [stac-utils/pystac#1375]
At some point in the near future version 1.1.0 of the STAC spec will be released. PRs are still being merged in, but the current list of high-profile changes is at: https://github.com/radiantearth/stac-spec/blob/v1.1.0-release/CHANGELOG.md 

These are the changes that I think will need to be made in pystac (in the order they appear in the stac-spec changelog):

- [ ] make bands core instead of being within the eo and raster extensions (WIP PR #1238)
- [ ] move `keywords` up out of collections into common metadata
- [ ] fix docstring for `license` to indicate that it supports `other` as an option and no longer supports `proprietary` and `various`.
- [ ] move `roles` up out of Assets and Providers and into common metadata
- [ ] add `collection` as a relation type
- [ ] deprecate `item_assets` extension and move into collection core
- [ ] #1347 
- [ ] in common metadata: validate that if a description is provided it is not empty
- [ ] ensure that APIs have trailing slashes and files do not (not sure this one requires any action)
- [ ] treat `start_datetime` and `end_datetime` as inclusive bounds
- [ ] `item.geometry` cannot be a  `GeometryCollection` (not sure this one requires any action)

Additionally pystac provides tools for migrating that walk the cataolog and migrate the STAC version and the extension versions. So those functions will need to be updated for this new version of STAC.

**Comment by jsignell** [comment]
Note: I ended up making a milestone for this and making all the items into individual tickets.

**Comment by gadomski** [comment]
@jsignell I'm starting work on v1.1.0 support, and I _think_ I'd like to get rid of the milestone and only use this ticket to track v1.1.0 issues. My reasoning is that some of these issues can be solved without breaking **pystac** (e.g. #1379) and others are breaking (e.g. changing the docstring for #1377 will inherently "move" **pystac** to v1.1.0). We'll also eventually need to update [DEFAULT_STAC_VERSION](https://github.com/stac-utils/pystac/blob/e326b41a86b308c49dce92f1a525fee05bb21928/pystac/version.py#L8-L9) which will be breaking as well.

Tl;dr: I'm proposing closing the `STAC spec version 1.0.0` milestone in favor of this tracking issue, and instead opening a `v1.11` milestone to go along with the existing [`v2.0`](https://github.com/stac-utils/pystac/milestone/28) milestone.

**Comment by jsignell** [comment]
> I'm proposing closing the STAC spec version 1.0.0 milestone in favor of this tracking issue, and instead opening a v1.11 milestone to go along with the existing [v2.0](https://github.com/stac-utils/pystac/milestone/28) milestone.

That plan sounds good to me! I felt a little wierd making a milestone that wasn't necessarily tied to a release.

**Comment by gadomski** [comment]
Ok, I've got things sorted into three milestones (I added one):

- I'd like to cut [v1.11](https://github.com/stac-utils/pystac/milestone/39) soon-ish to get it out _before_ we start in earnest on STAC v1.1 work
- [v1.12](https://github.com/stac-utils/pystac/milestone/38) is the (hopefully) non-breaking STAC v1.1 changes
- [v2.0](https://github.com/stac-utils/pystac/milestone/28) gets us home

**Comment by hrodmn** [comment]
Hi! I am working on a stactools package and am considering using `pystac>=1.12` but I want to be able to write items/collections that are compliant with STAC v1.0.0. I found the `set_stac_version` function but that doesn't seem to do much besides set the `stac_version` property in the items/collections that I create. Is it incumbent upon me to add the newly absorbed extensions (e.g. `item-assets`, `raster`) manually if I want to create items/collections that are compliant with v1.0.0?

**Comment by emmanuelmathot** [comment]
Wondering the status of this issue, in particular the bands construct support in `assets` and `item_assets`. I found this old issue: https://github.com/stac-utils/pystac/pull/1238. Any other current WIP?
